src/main/java/com/ee/employeesportal/controller/EmployeeController.java

```diff
@@ -2,47 +2,49 @@
 
 import com.ee.employeesportal.model.Employee;
 import com.ee.employeesportal.repositories.EmployeeRepository;
+import com.ee.employeesportal.services.EmployeeService;
 import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.domain.Sort.Direction;
+import org.springframework.data.domain.Sort.Order;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.ArrayList;
 import java.util.List;
 
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("/api/employees")
 public class EmployeeController {
-    private final EmployeeRepository employeeRepository;
 
-    @GetMapping("")
-    public List<Employee> getAllEmployees(){
-    return employeeRepository.getAllEmployees();
-    }
+    private final EmployeeService employeeService;
 
-    @GetMapping("/{empId}")
-    public ResponseEntity<Employee> getEmployeeById(@PathVariable(name = "empId") Long empId){
-        Employee employeeById = employeeRepository.getEmployeeById(empId);
-        if(employeeById != null)
-            return ResponseEntity.ok(employeeById);
-
-        return ResponseEntity.badRequest().build();
+    private Direction getSortDirection(String direction) {
+        if (direction.equals("asc")) {
+            return Direction.ASC;
+        } else if (direction.equals("desc")) {
+            return Direction.DESC;
+        }
+        return Direction.ASC;
     }
-
-    @PostMapping("")
-    public ResponseEntity<Employee> createEmployee(@RequestBody Employee employee){
-        employee = employeeRepository.createEmployee(employee);
-        return ResponseEntity.status(HttpStatus.CREATED).body(employee);
-    }
-
-    @PutMapping("/{empId}")
-    public Employee updateEmployeeById(@PathVariable(name = "empId") Long empId,@RequestBody Employee employee) {
-        employee.setEmpId(empId);
-        return employeeRepository.updateEmployeeById(employee);
+    @GetMapping("")
+    public List<Employee> getAllEmployees(){
+       return employeeService.getAllEmployees();
     }
 
-    @DeleteMapping("/{empId}")
-    public Employee deleteEmployeeById(@PathVariable(name = "empId") Long empId) {
-        return employeeRepository.deleteEmployeeById(empId);
+    @GetMapping("/sort")
+    public List<Employee> getAllEmployeeSortBy(@RequestParam List<String> sort_by){
```

Comment:
This can be compined with `/api/employees` API and pass sort parameters as query parameters.